### PR TITLE
gosec:chore - improve tests and clean code

### PR DESCRIPTION
**Pull request by matheusalcantarazup**
Previously the test cases of Gosec formatter was just testing that the
`StartAnalysis` method was not panicking and does not do any assert on
the results of analysis.

This commit improve the tests to assert that the results was parsed and
add correctly on Analysis.

This commit also make some clean up on code removing the unnecessary
package `entities` and move the schema of gosec output to gosec package
properly. Since these output schemas is not used by any package they was
changed to be private too.

Updates #718

Signed-off-by: Matheus Alcantara <matheus.alcantara@zup.com.br>

**- What I did**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: internal/services/formatters/go/gosec/formatter.go [hunk 1]
```diff
@@ -23,25 +23,45 @@ import (
...
+	Line       string                `json:"line"`
+	Column     string                `json:"column"`
+}
+
+// output represents the output schema from gosec.
+type output struct {
+	Issues []issue `json:"Issues"`
+}
+
+// Formatter represents the Gosec formatter.
```
Comment (iancardosozup) [inline internal/services/formatters/go/gosec/formatter.go:52]: i think there is no need for the entities package too, but i personally prefer a file for structs and other for implementation. What do you think to move the structs declaration to a file like `entities.go`?

Comment (matheusalcantarazup) [reply internal/services/formatters/go/gosec/formatter.go:52]: Make sense, I'll change this.

**Review by iancardosozup — APPROVED** [review]

File: internal/services/formatters/go/gosec/formatter_test.go [hunk 2]
```diff
@@ -24,112 +24,184 @@ import (
...
 	"github.com/ZupIT/horusec/internal/services/formatters"
 )
 
-func TestGoLang_StartAnalysis(t *testing.T) {
-	t.Run("Should run analysis and return error because output is empty", func(t *testing.T) {
-		dockerAPIControllerMock := &docker.Mock{}
-		dockerAPIControllerMock.On("SetAnalysisID")
-		dockerAPIControllerMock.On("CreateLanguageAnalysisContainer").Return("", nil)
+func TestGosecStartAnalysis(t *testing.T) {
+	t.Run("Should run analysis with sucessfull when output is empty", func(t *testing.T) {
```
Comment (nathanmartinszup) [inline internal/services/formatters/go/gosec/formatter_test.go:33]: I guess that `Should run analysis successfully when output is empty` is more accurate.
If decided to keep the message, there's a typo in `sucessfull` -> `successful`.

File: internal/services/formatters/go/gosec/formatter_test.go [hunk 3]
```diff
@@ -24,112 +24,184 @@ import (
...
+		docker := &docker.Mock{}
```
Comment (nathanmartinszup) [inline internal/services/formatters/go/gosec/formatter_test.go:34]: Docker variable contains the same imported package name.

File: internal/services/formatters/go/gosec/formatter_test.go [hunk 4]
```diff
@@ -24,112 +24,184 @@ import (
...
-			golangAnalyzer.StartAnalysis("")
-		})
+		NewFormatter(service).StartAnalysis("")
+		assert.False(t, entity.HasErrors(), "Expected no error for analysis")
 	})
 
-	t.Run("Should run analysis without panics and save on cache with success", func(t *testing.T) {
-		dockerAPIControllerMock := &docker.Mock{}
-		dockerAPIControllerMock.On("SetAnalysisID")
+	t.Run("Should run analysis with successful and add vulnerability on Analysis", func(t *testing.T) {
```
Comment (nathanmartinszup) [inline internal/services/formatters/go/gosec/formatter_test.go:47]: Same here `Should run analysis successfully and add vulnerability on Analysis`.

File: internal/services/formatters/go/gosec/formatter_test.go (hunk 2 above)
Comment (matheusalcantarazup) [reply internal/services/formatters/go/gosec/formatter_test.go]: Done.

File: internal/services/formatters/go/gosec/formatter_test.go (hunk 3 above)
Comment (matheusalcantarazup) [reply internal/services/formatters/go/gosec/formatter_test.go]: I don't see this as problem but I change.

File: internal/services/formatters/go/gosec/formatter_test.go (hunk 4 above)
Comment (matheusalcantarazup) [reply internal/services/formatters/go/gosec/formatter_test.go]: Done

File: internal/services/formatters/go/gosec/formatter_test.go (hunk 3 above)
Comment (nathanmartinszup) [reply internal/services/formatters/go/gosec/formatter_test.go]: Variables and packages with the same name may cause confusion. For example, trying to call a package function but it's calling the variable instead or inverse.

**Review by nathanmartinszup — APPROVED** [review]

**Pull request merged**